examples/helper_macros.rs

```diff
@@ -0,0 +1,27 @@
+//! A simple demonstration how iron's helper macros make e.g. IO-intensive code easier to write.
+#[macro_use] extern crate iron;
+
+use std::io;
+use std::fs;
+
+use iron::prelude::*;
+use iron::status;
+use iron::method;
+
+fn main() {
+    Iron::new(|req: &mut Request| {
+        Ok(match req.method {
+            method::Get => {
+                let f = itry!(fs::File::open("foo.txt"), (status::Ok, ""));
+                Response::with(status::Ok).set(f)
```

Comment:
possibly we should change the impl of Modifier for Path to send `NotFound` if the file doesn't exist, so this whole thing could be `Response::with((status::Ok, Path::new("foo.txt")))